src/schema/sources.ts

```diff
@@ -296,6 +296,10 @@ export const typeDefs = /* GraphQL */ `
       Commentary for the first share
       """
       commentary: String
+      """
+      Rank required for members to post
+      """
+      memberPostingRank: Int
```

Comment:
Curious if we should maybe expose this field as a string (SquadMemberRoles enum) and then just resolve integer here on the API side. Seems more bulletproof validation wise. WDYT?

Reply:
Concern is that we don't validate much so far so wanted to check first if some other reasons exist.

Reply:
Someone could theoretically send `999` through request.

Reply:
By requiring member name through string we would also be able to hide functionality of ranks from the frontend. Eg. frontend does not need to know that `owner` rank is `10`

Reply:
That makes sense, though this mutation is not exposed publicly and is a bit extreme edge case where they will not benefit from doing (hacking the submission) after going through. For now, I think this should be fine.

Reply:
Yes, this is exposed, but I mean **not** outside dailydev apps. We have CORS configured to only allow our known origins.

Reply:
He means in GitHub people can simply look up the code